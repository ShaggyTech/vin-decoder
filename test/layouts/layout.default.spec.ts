import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import defaultLayout from '@/layouts/default.vue';
import vuetifyConfig from '@/plugins/vuetify';

const localVue = createLocalVue();

jest.mock('@/compositions/useLocalStorage', () => ({
  syncHistoryOnMounted: () => jest.fn()
}));

const factory = (options: { [propName: string]: any } = {}) => {
  return mount(defaultLayout, {
    stubs: ['nuxt', 'router-link', 'AppTitleCard'],
    localVue,
    ...options
  });
};

describe('Layouts - default', () => {
  let vuetify;
  let wrapper: ReturnType<typeof factory>;

  beforeEach(() => {
    vuetify = new Vuetify(vuetifyConfig);
    wrapper = factory({
      vuetify
    });
  });

  test('component matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('component has "title" ref', () => {
    expect(wrapper.vm.$data.title).toBeDefined();
  });
});
