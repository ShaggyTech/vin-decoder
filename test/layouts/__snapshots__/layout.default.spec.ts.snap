// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts - default component matches snapshot 1`] = `
<div
  class="v-application v-application--is-ltr theme--dark"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <header
      class="v-sheet theme--dark v-toolbar v-app-bar v-app-bar--fixed"
      style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;"
    >
      <div
        class="v-toolbar__content"
        style="height: 64px;"
      >
        <div
          class="spacer"
        />
         
        <div
          class="v-toolbar__title display-1"
        >
          <nuxt-link-stub
            to="/"
          >
            <span
              class="primary_red__--text font-weight-black"
              id="header"
              innertext="VIN"
            >
              VIN
            </span>
             
            <span
              class="white__--text"
              id="subheader"
              innertext="Decoder"
            >
              Decoder
            </span>
          </nuxt-link-stub>
        </div>
         
        <div
          class="spacer"
        />
      </div>
    </header>
     
    <main
      class="v-main grey lighten-1"
      style="padding: 64px 0px 0px 0px;"
    >
      <div
        class="v-main__wrap"
      >
        <nuxt-stub />
      </div>
    </main>
     
    <footer
      class="v-footer text-center v-sheet theme--dark v-footer--fixed"
      style="left: 0px; right: 0px; bottom: 0px;"
    >
      <div
        class="text-center col col-12"
      >
        
      Made with ❤ by Brandon Eichler - ©
      2020
    
      </div>
    </footer>
  </div>
</div>
`;
